repose-aggregator/components/filters/body-patcher-filter/src/main/scala/org/openrepose/filters/bodypatcher/BodyPatcherFilter.scala

```diff
@@ -0,0 +1,150 @@
+  override val DEFAULT_CONFIG: String = "body-patcher.cfg.xml"
+  override val SCHEMA_LOCATION: String = "/META-INF/schema/config/body-patcher.xsd"
+
+  override def doWork(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+    val httpRequest: HttpServletRequest = request.asInstanceOf[HttpServletRequest]
+    val httpResponse: HttpServletResponse = response.asInstanceOf[HttpServletResponse]
+
+    val pathChanges: List[ChangeDetails] = filterChanges(httpRequest)
+    val requestPatches: List[Patch] = filterRequestChanges(pathChanges)
+    val responsePatches: List[Patch] = filterResponseChanges(pathChanges)
+
+    //patch the request
+    val patchedRequest: HttpServletRequest = determineContentType(httpRequest.getContentType) match {
+      case Json =>
+        val jsonPatches: List[String] = filterJsonPatches(requestPatches)
+        if (jsonPatches.nonEmpty) {
+          logger.debug("Applying json patches on the request")
+          val patchedValue: JsValue = applyJsonPatches(PJson.parse(httpRequest.getInputStream), jsonPatches)
+          new HttpServletRequestWrapper(httpRequest, new ServletInputStreamWrapper(new ByteArrayInputStream(PJson.stringify(patchedValue).getBytes)))
+        } else {
+          httpRequest
+        }
+      case Xml =>
+        //todo: implement xml support
+        httpRequest
+      case Other =>
+        httpRequest
+    }
+
+
+    //check if we might do work to the response
+    if (responsePatches.isEmpty) {
+      chain.doFilter(patchedRequest, httpResponse)
+    } else {
+      val wrappedResponse: HttpServletResponseWrapper = new HttpServletResponseWrapper(httpResponse, PASSTHROUGH, MUTABLE)
+      chain.doFilter(patchedRequest, wrappedResponse)
+      wrappedResponse.uncommit()
+
+      determineContentType(wrappedResponse.getContentType) match {
+        case Json =>
+          val jsonPatches: List[String] = filterJsonPatches(responsePatches)
+          if (jsonPatches.nonEmpty) {
+            logger.debug("Applying json patches on the response")
+
+            val contentStream = wrappedResponse.getOutputStreamAsInputStream
+            val contentValue = if (contentStream.available() > 0) {
+              PJson.parse(contentStream)
+            } else {
+              PJson.obj()
+            }
+            val patchedValue: JsValue = applyJsonPatches(contentValue, jsonPatches)
+            wrappedResponse.setOutput(new ByteArrayInputStream(PJson.stringify(patchedValue).getBytes))
+          }
+        case Xml => //todo implement xml support
+        case Other =>
+      }
+      wrappedResponse.commitToResponse()
+    }
+  }
+
+  def filterChanges(request: HttpServletRequest): List[ChangeDetails] = {
+    val urlPath: String = new URL(request.getRequestURL.toString).getPath
+    configuration.getChange.asScala.toList
+        .filter(_.getPath.r.findFirstIn(urlPath).isDefined)
```

Comment:
I should note that this approach to regex matching only requires that the configured regex matches some portion of the actual string. It's an issue that has come up before.

For example, if I've configured a change to occur on `http://example.com/` and the request URI is `http://example.com/foo/bar` I might not expect the change to occur, but it will. Of course, that can be remedied by explicitly included start of string or end of string matchers, like so: `^http://example.com/$`. 
```
scala> "/".r.findFirstIn("/foo").isDefined
res1: Boolean = true

scala> "^/$".r.findFirstIn("/foo").isDefined
res3: Boolean = false
```

The alternative would be to turn the Scala regex into a Java pattern and call `matches` which attempts a full match of the regex against the string.
```
scala> "/".r.pattern.matcher("/foo").matches
res2: Boolean = false
```

Ultimately, it just depends what we want the user experience to be.